### PR TITLE
AF-2215: Avoid losing scroll position when editor changes

### PR DESCRIPTION
**Pull request by jesuino**
This solution for AF-2215 records the scroll position and restore it after the layout editor draw all layout elements.

This should be tested with Form Editor and Page Layout Editor.

### CODE REVIEW

**Comment by jesuino** [comment]
@ederign would you please review?

@domhanak  do you know who could review this?

Thanks!

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/infra/LayoutEditorFocusController.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import org.uberfire.ext.layout.editor.client.components.container.Container;
+
+import com.google.gwt.core.client.Scheduler;
+import com.google.gwt.dom.client.Style;
+import com.google.gwt.dom.client.Style.Overflow;
+
+import elemental2.dom.HTMLElement;
+import jsinterop.base.Js;
+
+@ApplicationScoped
```
Comment (paulovmr) [inline uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/infra/LayoutEditorFocusController.java]: Could you move this above the class name?

Comment (jesuino) [reply uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/infra/LayoutEditorFocusController.java]: oops, done, thanks @paulovmr

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Comment by dgutierr** [comment]
@jesuino mind to add Liz as a reviewer please and not merge under she approves?

**Comment by lclay2** [comment]
There's no way for UX to verify this. Could you please post a video demonstrating the fix.

**Comment by lclay2** [comment]
Based on the video I saw, it appears to be resolved. The form stays in the same position.

**Comment by paulovmr** [comment]
Unrelated test failure

**Pull request merged**